x/ccv/parent/keeper/relay.go

```diff
@@ -113,7 +118,57 @@ func (k Keeper) EndBlockCallback(ctx sdk.Context) {
 		if len(valUpdates) != 0 {
 			k.SendPacket(ctx, chainID, valUpdates, valUpdateID)
 			k.IncrementValidatorSetUpdateId(ctx) // TODO: this needs to be moved out of this scope
+			k.SetValsetUpdateBlockHeight(ctx, valUpdateID, uint64(ctx.BlockHeight()))
 		}
 		return false
 	})
 }
+
+// OnRecvPacket slahes and jails the given validator in the packet data
+func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data ccv.ValidatorDowntimePacketData) exported.Acknowledgement {
+	if childChannel, ok := k.GetChannelToChain(ctx, packet.DestinationChannel); !ok && childChannel != packet.DestinationChannel {
+		ack := channeltypes.NewErrorAcknowledgement(
+			fmt.Sprintf("packet sent on a channel %s other than the established child channel %s", packet.DestinationChannel, childChannel),
+		)
+		chanCap, _ := k.scopedKeeper.GetCapability(ctx, host.ChannelCapabilityPath(packet.DestinationPort, packet.DestinationChannel))
+		k.channelKeeper.ChanCloseInit(ctx, packet.DestinationPort, packet.DestinationChannel, chanCap)
+		return &ack
+	}
+
+	// initiate slashing and jailing
+	if err := k.HandleConsumerDowntime(ctx, data); err != nil {
+		ack := channeltypes.NewErrorAcknowledgement(err.Error())
+		return &ack
+	}
+
+	return nil
```

Comment:
Is it mandatory to send an ack? It seems like it can't be because we don't send an ack for 2 weeks in the core protocol, and around 345k packets could be received without an ack coming back. 

Sending it just means that we have more boilerplate on the other side.

Reply:
From  [ICS 4](https://github.com/cosmos/ibc/tree/master/spec/core/ics-004-channel-and-packet-semantics#writing-acknowledgements):
> This is an asynchronous acknowledgement, the contents of which do not need to be determined when the packet is received, only when processing is complete.
This is why the consumer can receive around 345k packets w/o ACKs. However, no ACKs can be skipped (see below).

> Acknowledging packets is not required; however, if an ordered channel uses acknowledgements, either all or no packets must be acknowledged (since the acknowledgements are processed in order). 

I guess the meaning here is a uni-directional channel, right? I think the packets sent by the consumer do not need to be ACKed.

> Note that if packets are not acknowledged, packet commitments cannot be deleted on the source chain. Future versions of IBC may include ways for modules to specify whether or not they will be acknowledging packets in order to allow for cleanup.

However, to enable cleanup, we may need to send ACKs for all packets.